### PR TITLE
Digests must be unique in manifest and fixity

### PR DESCRIPTION
**Pull request by neilsjefferies**
Addresses #461

### CODE REVIEW

File: draft/spec/index.html [hunk 1]
```diff
@@ -654,7 +654,9 @@ <h2>Manifest</h2>
                 of every content file in all versions of the <a>OCFL Object</a>. The value for each key
                 <span id="E092">MUST</span> be an array containing the <a>content path</a>s of files in the OCFL
                 Object that have content with the given digest. Content paths within a manifest block
-                <span id="E042">MUST</span> be relative to the <a>OCFL Object Root</a>.
+                <span id="E042">MUST</span> be relative to the <a>OCFL Object Root</a>. As JSON keys are case 
+                sensitive, while digests may not be, there is an additional requirement that each digest value
```
Comment (ahankinson) [inline draft/spec/index.html:658]: "may not be" --> "are not" since we specify that digests are case insensitive.

Comment (neilsjefferies) [reply draft/spec/index.html:658]: Not all digests are ncessarily case insensitive and we don't actually require it as far as I can see. We just note that it is the case for the base16 digests that we currently specify.

Comment (zimeon) [reply draft/spec/index.html:658]: I think "may not be" is correct. Some digests might be case sensitive even if that doesn't apply to the ones we have specified so far. I would leave it as is.

**Comment by zimeon** [comment]
This PR seems good as far as it goes but there are two more issues that it does not address:
* Should a digest also appear only once in a `state` block?
* I assume the sense is that the case in `manifest` and case in `state` do not have to match, I think we should note that

**Comment by zimeon** [comment]
We could always merge this PR and have a new PR for the other parts, but I think they are still part of #461

File: draft/spec/index.html (hunk 1 above)
Comment (neilsjefferies) [reply draft/spec/index.html:658]: ...but it IS awkward to validate if we can't assume that digests are case insensitive. Which would mean modifying the language around digests.

Comment (zimeon) [reply draft/spec/index.html:658]: IMO, every digest needs its own normalization scheme. That is what I built into my code

File: validation/validation-codes.md [hunk 2]
```diff
@@ -82,6 +82,8 @@
 | E093 | 'Where included in the fixity block, the digest values given must match the digests of the files at the corresponding content paths.' | https://ocfl.io/draft/spec/#E093
 | E094 | 'The value of [the message] key is freeform text, used to record the rationale for creating this version. It must be a JSON string.' | https://ocfl.io/draft/spec/#E094
 | E095 | 'Within a version, logical paths must be unique and non-conflicting, so the logical path for a file cannot appear as the initial part of another logical path.' | https://ocfl.io/draft/spec/#E095
+| E096 | 'As JSON keys are case sensitive, while digests may not be, there is an additional requirement that each digest value must occur only once in the manifest regardless of case.' | https://ocfl.io/draft/spec/#E096
+| E097 | 'As JSON keys are case sensitive, while digests may not be, there is an additional requirement that each digest value must occur only once in the fixity block for any digiest algorithm, regardless of case.' | https://ocfl.io/draft/spec/#E097
```
Comment (ahankinson) [inline validation/validation-codes.md:86]: typo "digest"

**Comment by awoods** [comment]
@zimeon : Regarding:
> Should a digest also appear only once in a state block?

Given the statement in the `state` block description:
> The keys of this JSON object are digest values, each of which must correspond to an entry in the manifest of the inventory.

..maybe it is a non-issue? We could clarify what we mean by "correspond" to note that the case must match.

**Comment by zimeon** [comment]
@awoods I think we need to be more specific that the keys in `state` must be exact string matches to the digest values in the `manifest`

File: validation/validation-codes.md (hunk 2 above)
Comment (neilsjefferies) [reply validation/validation-codes.md]: Good spot! Can't even cut-and-paste competently!

File: draft/spec/index.html [hunk 3]
```diff
@@ -654,7 +654,9 @@ <h2>Manifest</h2>
                 of every content file in all versions of the <a>OCFL Object</a>. The value for each key
                 <span id="E092">MUST</span> be an array containing the <a>content path</a>s of files in the OCFL
                 Object that have content with the given digest. Content paths within a manifest block
-                <span id="E042">MUST</span> be relative to the <a>OCFL Object Root</a>.
+                <span id="E042">MUST</span> be relative to the <a>OCFL Object Root</a>. As JSON keys are case
```
[the remark this answers is not in the document]
Comment (neilsjefferies) [reply draft/spec/index.html]: Agreed, and something similar for the fixity section too.

**Pull request merged**